design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,192 @@
+
+#### From Scratch (Top Navigation)
+
+- In the `Schema Browser` view, a user clicks on the `Data Explorer` navigation item located in the top navigation bar.
+- `Data Explorer` opens, and the layout changes. The schema explorer sidebar is hidden, as well as any open table.
+
+#### From a Table (Sidebar Navigation or Open Table)
+
+- From an open table, a user clicks the `Create View from [Table Name]` button located in the toolbar. This action will open `Data Explorer` and automatically set the table as `Base Table` and select all direct columns.
+- In the `Schema Browser` view, a user opens the contextual menu for a table navigation item and selects `Create view from Table`. This action launches `Data Explorer` and automatically sets the selected table as `Base Table` and adds its direct columns to a `Select Column` step.
+- The system will also add link columns if present in the `Base Table`. However, it won't include link columns referencing the `Base Table` from other tables. The system will set the values for link columns to the referenced table's primary key by default.
+- The user can then choose to remove or add additional columns.
+
+### User opens the Data Explorer to inspect an existing view
+
+#### From an open view
+
+- From an active view tab, a user can inspect a view by clicking on the `Open in Data Explorer` button in the toolbar area.
+- Views need to use supported query commands and functions to be inspected in `Data Explorer` with full functionality. When queries are not supported, an option to see the underlying query will be available.
+
+#### From a view navigation item
+
+- In the `Schema Browser` view, a user opens the contextual menu for a view navigation item and selects the option `Open in Data Explorer`. This action will open `Data Explorer` and display the properties of the selected view.
+
+## Base Table
+
+Views require a base table to be set so that the available columns and links can be listed for selection.
+
+### User sets the base table for a new view
+
+- For users starting a new view from scratch, `Base Table` selection will be the first required step.
+- Once `Data Explorer` is launched the user will be prompted to make a selection. The system will list all tables from the active schema. See *Wireframe 1* for an example of the base table selection prompt.
+- Once the `Base Table` is selected, a second step for `Select Columns` is automatically added and the user is prompted to continue with the view creation process.
+- At this point, the result table is empty.
+
+[Wireframe 1](https://share.balsamiq.com/c/jp7WFu663j6sbfKNknuq5Z.png)
+
+### User inspects the base table for an existing view in Data Explorer
+
+- When inspecting an existing view, a user can find the `Base Table` as the first step of the `Workflow Sidebar`.
+- The `Base Table` step cannot be removed, only edited, causing all subsequent steps to be cleared.
+
+[Wireframe 2](https://share.balsamiq.com/c/wTEzQcBamoBBWBSSF5Cmuy.png)
```

Comment:
I'm wondering if we need this to be an explicit "step" given that the user cannot remove it. Here is a rough idea:

![2 - Base Table Columns](https://user-images.githubusercontent.com/287034/158439130-46880505-8d4e-4b56-b4db-f9cadeb1b780.png)

Reasoning:
- I don't think we need the user to learn the term "base table" I'd like to have an interface where the user needs to learn as few terms as possible. 
- I think this also conveys that the base table affects the entire exploration (query), which is something we want users to learn.